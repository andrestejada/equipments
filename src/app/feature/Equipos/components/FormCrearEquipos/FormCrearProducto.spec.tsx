import '@testing-library/jest-dom/extend-expect';
import { fireEvent, prettyDOM, render } from '@testing-library/react';
import { FormCrearEquipos } from './index';
import React from 'react';
import { newEquipment } from '../../../../shared/fixture/testData';

const addNewEquipment= jest.fn();

describe('Name of the group', () => {
    let wrapperComponent = render(
        <FormCrearEquipos
            addNewEquipment={addNewEquipment}
        />
    );
    beforeEach(()=>{
        wrapperComponent = render(
            <FormCrearEquipos
                addNewEquipment={addNewEquipment}
            />
        );
        jest.clearAllMocks();
    });
    it('should render corrrectly', () => {
        expect(wrapperComponent).toMatchSnapshot();
    });

 
    it('should be find the alert when press submit button', async() => {
        const button = wrapperComponent.getByText('Ingresar');
        fireEvent.click(button);  
        const errors = await  wrapperComponent.findAllByText(/requerid/i);
        expect(errors[0]).toBeInTheDocument();
        expect(errors[0].textContent).toBe('El codigo es requerido');
        expect(errors[1]).toBeInTheDocument();
        expect(errors[1].textContent).toBe('El nombre es requerido');
        expect(errors[2]).toBeInTheDocument();
        expect(errors[2].textContent).toBe('La ubicacion es requerida');
    });

    it('the form should be submit correctly', async() => {
        const {codigo,nombre,ubicacion} = newEquipment;
            const button = wrapperComponent.getByText('Ingresar');
            fireEvent.change(wrapperComponent.getByPlaceholderText(/ingresa un codigo/i),{target:{value:codigo}});
            fireEvent.change(wrapperComponent.getByPlaceholderText(/ingresa un nombre/i),{target:{value:nombre}});
            fireEvent.change(wrapperComponent.getByRole(/ubicacion/i),{target:{value:ubicacion}});
            fireEvent.click(button);  
            const codeValue = await wrapperComponent.findByPlaceholderText(/ingresa un codigo/i); 
            expect(codeValue).toHaveTextContent('');            
            expect(addNewEquipment).toHaveBeenCalled();        
            expect(addNewEquipment).toHaveBeenCalledWith(newEquipment);   
               
    });
});