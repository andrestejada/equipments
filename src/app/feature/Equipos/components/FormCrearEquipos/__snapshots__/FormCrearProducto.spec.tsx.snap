// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name of the group should render corrrectly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2
          style="text-align: center; margin: 0.8rem 0px;"
        >
          Empieza Creando Un Equipo
        </h2>
        <form
          class="sc-AxhCb gShkfD"
          role="form"
        >
          <h3
            style="margin: 0.8rem 0px;"
          >
            Ingresar los datos
          </h3>
          <hr />
          <div
            class="sc-AxhUy feubkZ"
          >
            <label
              class="sc-AxheI hzItYT"
            >
              Codigo
            </label>
            <input
              class="sc-AxgMl eKBvHB"
              name="codigo"
              placeholder="Ingresa un codigo"
              type="number"
              value=""
            />
          </div>
          <div
            class="sc-AxhUy feubkZ"
          >
            <label
              class="sc-AxheI hzItYT"
            >
              Nombre Equipo
            </label>
            <input
              class="sc-AxgMl eKBvHB"
              name="nombre"
              placeholder="Ingresa un nombre para el equipo"
              type="text"
              value=""
            />
          </div>
          <div
            class="sc-AxhUy feubkZ"
          >
            <label
              class="sc-AxheI hzItYT"
            >
              Ubicacion
            </label>
            <select
              class="sc-Axmtr fHwtdx"
              name="ubicacion"
              role="ubicacion"
            >
              <option
                hidden=""
                value=""
              >
                Selecciona una ubicación
              </option>
              <option
                value="laboratorio"
              >
                Laboratorio
              </option>
              <option
                value="fabrica"
              >
                Fabrica
              </option>
              <option
                value="almacen"
              >
                Almacen
              </option>
            </select>
          </div>
          <button
            class="sc-AxiKw eJVtzL"
            role="button"
            type="submit"
          >
            Ingresar
          </button>
        </form>
      </div>
    </div>
    <div>
      <div>
        <h2
          style="text-align: center; margin: 0.8rem 0px;"
        >
          Empieza Creando Un Equipo
        </h2>
        <form
          class="sc-AxhCb gShkfD"
          role="form"
        >
          <h3
            style="margin: 0.8rem 0px;"
          >
            Ingresar los datos
          </h3>
          <hr />
          <div
            class="sc-AxhUy feubkZ"
          >
            <label
              class="sc-AxheI hzItYT"
            >
              Codigo
            </label>
            <input
              class="sc-AxgMl eKBvHB"
              name="codigo"
              placeholder="Ingresa un codigo"
              type="number"
              value=""
            />
          </div>
          <div
            class="sc-AxhUy feubkZ"
          >
            <label
              class="sc-AxheI hzItYT"
            >
              Nombre Equipo
            </label>
            <input
              class="sc-AxgMl eKBvHB"
              name="nombre"
              placeholder="Ingresa un nombre para el equipo"
              type="text"
              value=""
            />
          </div>
          <div
            class="sc-AxhUy feubkZ"
          >
            <label
              class="sc-AxheI hzItYT"
            >
              Ubicacion
            </label>
            <select
              class="sc-Axmtr fHwtdx"
              name="ubicacion"
              role="ubicacion"
            >
              <option
                hidden=""
                value=""
              >
                Selecciona una ubicación
              </option>
              <option
                value="laboratorio"
              >
                Laboratorio
              </option>
              <option
                value="fabrica"
              >
                Fabrica
              </option>
              <option
                value="almacen"
              >
                Almacen
              </option>
            </select>
          </div>
          <button
            class="sc-AxiKw eJVtzL"
            role="button"
            type="submit"
          >
            Ingresar
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2
        style="text-align: center; margin: 0.8rem 0px;"
      >
        Empieza Creando Un Equipo
      </h2>
      <form
        class="sc-AxhCb gShkfD"
        role="form"
      >
        <h3
          style="margin: 0.8rem 0px;"
        >
          Ingresar los datos
        </h3>
        <hr />
        <div
          class="sc-AxhUy feubkZ"
        >
          <label
            class="sc-AxheI hzItYT"
          >
            Codigo
          </label>
          <input
            class="sc-AxgMl eKBvHB"
            name="codigo"
            placeholder="Ingresa un codigo"
            type="number"
            value=""
          />
        </div>
        <div
          class="sc-AxhUy feubkZ"
        >
          <label
            class="sc-AxheI hzItYT"
          >
            Nombre Equipo
          </label>
          <input
            class="sc-AxgMl eKBvHB"
            name="nombre"
            placeholder="Ingresa un nombre para el equipo"
            type="text"
            value=""
          />
        </div>
        <div
          class="sc-AxhUy feubkZ"
        >
          <label
            class="sc-AxheI hzItYT"
          >
            Ubicacion
          </label>
          <select
            class="sc-Axmtr fHwtdx"
            name="ubicacion"
            role="ubicacion"
          >
            <option
              hidden=""
              value=""
            >
              Selecciona una ubicación
            </option>
            <option
              value="laboratorio"
            >
              Laboratorio
            </option>
            <option
              value="fabrica"
            >
              Fabrica
            </option>
            <option
              value="almacen"
            >
              Almacen
            </option>
          </select>
        </div>
        <button
          class="sc-AxiKw eJVtzL"
          role="button"
          type="submit"
        >
          Ingresar
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
